src/vs/workbench/contrib/chat/browser/media/chatSessions.css

```diff
@@ -169,7 +169,17 @@
 	text-align: center;
 }
 
-.chat-sessions-tree-container .chat-session-item .monaco-icon-label.codicon-loading::before {
+/* Custom chat session icon that bypasses file icon theme */
+.chat-sessions-tree-container .chat-session-item .chat-session-custom-icon {
+	flex-shrink: 0;
+	width: 16px;
+	height: 16px;
+	margin-right: 6px;
+	font-size: 16px;
+	color: var(--vscode-icon-foreground);
+}
+
+.chat-sessions-tree-container .chat-session-item .chat-session-custom-icon.codicon-loading::before {
```

Comment:
Could you try changing [this loading icon](https://github.com/microsoft/vscode/blob/fc4d4d8ee53e384002f341411c71326a4fddfc51/src/vs/workbench/contrib/chat/browser/chatSessions/view/sessionsTreeRenderer.ts#L220-L221) to be `ThemeIcon.modify(Codicon.loading, 'spin')` and see if that works now that we are using IconLabel? 
If it does then we can remove this CSS rule